src/fides/api/task/task_resources.py

```diff
@@ -147,7 +147,22 @@ def get_all_cached_objects(self) -> Dict[str, Optional[List[Row]]]:
             f"{self.request.id}__access_request"
         )
         # extract request id to return a map of address:value
-        return {k.split("__")[-1]: v for k, v in value_dict.items()}
+        return {self.extract_key_for_address(k): v for k, v in value_dict.items()}
+
+    def extract_key_for_address(self, full_request_id: str) -> str:
+        """
+        Handles extracting the correct Dataset:Collection to map to extracted
+        values.
+
+        Handles an edge case where double underscores exist in either the fides_key
+        of the Dataset or the Collection name.
+        """
+        request_id_dataset, collection = full_request_id.split(":")
+        number_of_expected_items_to_remove = 2
+        request_list = request_id_dataset.split("__")
+        address_list = request_list[number_of_expected_items_to_remove:]
+        dataset = "__".join(address_list)
```

Comment:
this looks like it should get the job done, but i _think_ we can use the `maxsplit` arg of `split()` to make this a bit more concise/straightforward. would something like the below work? it seems to pass the updated test.

```suggestion
        dataset = request_id_dataset.split("__", 2)[-1]
```

(may need some a comment or two to clarify why we're splitting this way)

Reply:
Ya I think your idea here makes sense 👍🏽 

My thoughts here were around trying to avoid a magic number scenario like we hit with all of the custom fields changes by being overly verbose with what each step was doing and why (thinking it easier to debug in the future if/when we hit another edge case)

The only other change I might make here is keeping `number_of_expected_items_to_remove` to self document why the number 2 is there if that works?

Reply:
Fixed in https://github.com/ethyca/fides/pull/3688/commits/7371272ecaec02b7a52df15286a83a668db62d60

Reply:
very nice, i like where you landed!